sdk/enoki/demo/App.tsx

```diff
@@ -1,40 +1,41 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { getFullnodeUrl, SuiClient } from '@mysten/sui/client';
+import {
+	ConnectButton,
+	useConnectWallet,
+	useCurrentAccount,
+	useSignAndExecuteTransaction,
+} from '@mysten/dapp-kit';
 import { Transaction } from '@mysten/sui/transactions';
 import { useState } from 'react';
 
-import { useAuthCallback, useEnokiFlow, useZkLogin } from '../src/react.tsx';
+import { useEnokiWallets } from '../src/react.js';
+import { EnokiWallet } from '../src/wallet/index.js';
 
 export function App() {
-	const flow = useEnokiFlow();
-	const zkLogin = useZkLogin();
-	const [result, setResult] = useState<any>(null);
+	const { mutate: connect } = useConnectWallet();
+	const currentAccount = useCurrentAccount();
+	const [result, setResult] = useState<any>();
 
-	useAuthCallback();
+	const { wallets, execute } = useEnokiWallets();
+	const { mutateAsync: signAndExecute } = useSignAndExecuteTransaction({
+		execute,
+	});
 
 	return (
 		<div>
-			<div>Address: {zkLogin.address}</div>
-			<div>Provider: {zkLogin.provider}</div>
-			{!zkLogin.address ? (
-				<button
-					onClick={async () => {
-						window.location.href = await flow.createAuthorizationURL({
-							provider: 'google',
-							clientId: '705781974144-cltddr1ggjnuc3kaimtc881r2n5bderc.apps.googleusercontent.com',
-							redirectUrl: window.location.href.split('#')[0],
-						});
-					}}
-				>
-					Sign in with Google
-				</button>
-			) : (
-				<button onClick={() => flow.logout()}>Sign Out</button>
-			)}
+			<ConnectButton walletFilter={(wallet) => !(wallet instanceof EnokiWallet)} />
```

Comment:
Two things:
- I wonder if there's a better way to do this instead of using `instanceof`.
- I think we should _potentially_ do this by default in dapp-kit (so that enoki wallets are filtered out).

Reply:
I added an isEnokiWallet function that uses the wallet id, which is prefixed with `enoki:`.  I am not sure filtering by default in dapp-kit is a good idea, but that might be more of a product question.  from a technical perspective I don't like it